parallelcluster/add-external-lb.sh

```diff
@@ -1,18 +1,43 @@
 #!/bin/bash
-export cluster="security3"
-tempfile=`mktemp`
-aws elbv2 describe-tags --resource-arns $(aws elbv2 describe-target-groups --query 'TargetGroups[*].TargetGroupArn' --output text) --query "TagDescriptions[?Tags[?Key==\`parallelcluster:cluster-name\` && Value==\`$cluster\`]].ResourceArn" --output text > $tempfile
-target_group_arn=`cat $tempfile`
+export cluster="security"
+
+# Find existing internal LB ARN
+tempfile=$(mktemp)
+aws elbv2 describe-tags --resource-arns $(aws elbv2 describe-load-balancers --query 'LoadBalancers[*].LoadBalancerArn' --output text) \
+--query "TagDescriptions[?Tags[?Key=='parallelcluster:cluster-name' && Value=='$cluster']].ResourceArn" --output text > $tempfile
+internal_lb_arn=$(cat $tempfile)
+rm $tempfile
+
+# DNS Name of internal LB 
+internal_lb_dns=`aws elbv2 describe-load-balancers --load-balancer-arns $internal_lb_arn --query 'LoadBalancers[0].DNSName' --output text`
+
+# We need the head node id to inquire the IP address of the internal LB (the AWS API does not seem to expose the IP of a NLB)
+head_node_id=`pcluster describe-cluster --cluster-name="$cluster" | jq -r .headNode.instanceId`
```

Comment:
I'm surprised, but you're right. The IP of the internal nlb is not available via the AWS API. But it seems that you can get the IP by just locally running `dig +short securi-secur-Q2KE3WI0y9Xv-04f0c6047f302af4.elb.eu-west-1.amazonaws.com`

I think it may make sense to cut out some of this complexity.

Reply:
Note: if this doesn't work for you, then I may have borked my local DNS setup. 🤷

Reply:
> I'm surprised, but you're right. The IP of the internal nlb is not available via the AWS API. But it seems that you can get the IP by just locally running `dig +short securi-secur-Q2KE3WI0y9Xv-04f0c6047f302af4.elb.eu-west-1.amazonaws.com`
> 
> I think it may make sense to cut out some of this complexity.

Completely agreed, but in my "vanilla" network setup on my iMac `dig +short securi-secur-Q2KE3WI0y9Xv-04f0c6047f302af4.elb.eu-west-1.amazonaws.com` just hangs, hence the extra complexity...

Reply:
According to ChatGPT: "If you're trying to resolve the IP addresses of a private NLB from outside the VPC (e.g., from your local laptop), you won't be able to directly resolve the DNS name to IP addresses because private NLBs are not accessible from the public internet. They are designed to be accessible only within the VPC." 

Arguably however we could run this command remotely from the jump host as well, but then we need to be careful with IAM permissions and policies again...